samples/Echo_Ssl/app/application.cpp

```diff
@@ -25,51 +26,13 @@ TcpClient* client;
 bool showMeta = true;
 
 /* Debug SSL functions */
-void displaySessionId(SSL* ssl)
+void displaySessionId(SslSessionId* sessionId)
 {
-	int i;
-	const uint8_t* session_id = ssl_get_session_id(ssl);
-	int sess_id_size = ssl_get_session_id_size(ssl);
-
-	if(sess_id_size > 0) {
-		debugf("-----BEGIN SSL SESSION PARAMETERS-----\n");
-		for(i = 0; i < sess_id_size; i++) {
-			debugf("%02x", session_id[i]);
-		}
-
-		debugf("\n-----END SSL SESSION PARAMETERS-----\n");
-	}
-}
-
-/**
- * Display what cipher we are using
- */
-void displayCipher(SSL* ssl)
-{
-	Serial.printf("CIPHER is ");
-	switch(ssl_get_cipher_id(ssl)) {
-	case SSL_AES128_SHA:
-		Serial.printf("AES128-SHA");
-		break;
-
-	case SSL_AES256_SHA:
-		Serial.printf("AES256-SHA");
-		break;
-
-	case SSL_AES128_SHA256:
-		Serial.printf("SSL_AES128_SHA256");
-		break;
-
-	case SSL_AES256_SHA256:
-		Serial.printf("SSL_AES256_SHA256");
-		break;
-
-	default:
-		Serial.printf("Unknown - %d", ssl_get_cipher_id(ssl));
-		break;
+	if(sessionId != nullptr && sessionId->isValid()) {
+		debugf("-----BEGIN SSL SESSION PARAMETERS-----");
+		debugf("%s", makeHexString(sessionId->getValue(), sessionId->getLength()).c_str());
+		debugf("-----END SSL SESSION PARAMETERS-----");
```

Comment:
Maybe add some of this debug stuff to the framework using `Print&` instead of debug functions.